.github/workflows/ci_linux_x64_asan-libshortfin.yml

```diff
@@ -0,0 +1,151 @@
+
+    - name: Checkout pyenv
+      if: steps.cache-python-asan.outputs.cache-hit != 'true'
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      with:
+        repository: pyenv/pyenv
+        ref: ${{ env.PYENV_REF }}
+        path: ${{ env.PYENV_DIR }}
+
+    - name: Install pyenv & Python
+      if: steps.cache-python-asan.outputs.cache-hit != 'true'
+      run: |
+        cd ${{ env.PYENV_DIR }}
+        src/configure && make -C src
+        export PYENV_ROOT=${{ env.PYENV_DIR }}
+        export PATH=${{ env.PYENV_DIR }}/bin:$PATH && eval "$(pyenv init -)"
+        CC=clang-18 CXX=clang++-18 LDFLAGS="-lstdc++" PYTHON_CONFIGURE_OPTS="--with-address-sanitizer" pyenv install -v -g ${{ env.PYTHON_VER }}
+
+
+  build-and-test:
+    name: Build and test libshortfin
+    needs: [setup-python-asan]
+    runs-on: ubuntu-24.04
+
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install clang lld cmake ninja-build
+
+    - name: Checkout repository
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      with:
+        submodules: false
+
+    - name: Checkout IREE repo
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      with:
+        repository: iree-org/iree
+        path: ${{ env.IREE_SOURCE_DIR }}
+        submodules: false
+
+    - name: Initalize IREE submodules
+      run : |
+        cd ${{ env.IREE_SOURCE_DIR }}
+        git submodule update --init --depth 1 -- third_party/benchmark
+        git submodule update --init --depth 1 -- third_party/cpuinfo/
+        git submodule update --init --depth 1 -- third_party/flatcc
+        git submodule update --init --depth 1 -- third_party/googletest
+        git submodule update --init --depth 1 -- third_party/hip-build-deps/
+
+    - name: Setup python
+      id: cache-python-asan
+      uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: ${{ env.PYENV_DIR }}
+        key: ${{ runner.os }}-python-asan-${{ env.PYENV_REF }}-${{ env.PYTHON_VER }}-${{ env.CACHE_VER }}
+    - name: Install Python packages
+      # TODO: Switch to `pip install -r requirements.txt -e libshortfin/`.
+      run: |
+        export PYENV_ROOT=${{ env.PYENV_DIR }}
+        export PATH=${{ env.PYENV_DIR }}/bin:$PATH && eval "$(pyenv init -)"
+        pyenv versions
+        pyenv global ${{ env.PYTHON_VER }}-debug
+        pyenv local ${{ env.PYTHON_VER }}-debug
+        pyenv shell ${{ env.PYTHON_VER }}-debug
+        pip install nanobind
+        pip install -r ${{ env.LIBSHORTFIN_DIR }}/requirements-tests.txt
+
+    - name: Build libshortfin
+      run: |
+        export PYENV_ROOT=${{ env.PYENV_DIR }}
+        export PATH=${{ env.PYENV_DIR }}/bin:$PATH && eval "$(pyenv init -)"
+        pyenv global ${{ env.PYTHON_VER }}-debug
```

Comment:
I don't think this is necessary to do again as it is changing symlinks and such, not mucking with the environment.

Reply:
I have cut it down to what is strictly needed.